### PR TITLE
[Snyk] Security upgrade eslint-config-expensify from 2.0.42 to 2.0.44

### PR DESCRIPTION
**Pull request by MelvinBot**
<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to fix one or more vulnerable packages in the `npm` dependencies of this project.</h3>

#### Changes included in this PR

- Changes to the following files to upgrade the vulnerable dependencies to a fixed version:
    - package.json
    - package-lock.json

##### With an upgrade:
Severity                   | Priority Score (*)                   | Issue                   | Breaking Change                   | Exploit Maturity
:-------------------------:|-------------------------|:-------------------------|:-------------------------|:-------------------------
high severity  |  **696/1000**  <br/> **Why?** Proof of Concept exploit, Has a fix available, CVSS 7.5  | Regular Expression Denial of Service (ReDoS) <br/>[SNYK-JS-ANSIREGEX-1583908](https://snyk.io/vuln/SNYK-JS-ANSIREGEX-1583908) |  No  | Proof of Concept 

(*) Note that the real score may have changed since the PR was raised.

<details>
  <summary><b>Commit messages</b></summary>
  </br>
  <details>
    <summary>Package name: <b>eslint-config-expensify</b></summary>
    The new version differs by 15 commits.</br>
    <ul>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/72d77f0eae7e66775b98ded3e9c7f7b0a6144dc4">72d77f0</a> Merge pull request #86 from Expensify/OSBotify-bump-version-9ba6309f-decf-418e-9315-d283acdcef96</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/c321591ba76d421e6ac5e61bf8717c18b9bf8030">c321591</a> 2.0.44</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/a15f930699c3149eee46c2df5391620ccc75195b">a15f930</a> Merge pull request #85 from blazejkustra/react-hooks/recommended</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/ec28d49b4383ae55c66981da8a421326c4519914">ec28d49</a> Downgrade some packages</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/94b5c77c5250ed141b47af6fbdd2791831111834">94b5c77</a> Downgrade jest</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/62d889dbd1b92ea5c04d32441339e18be1826a92">62d889d</a> Fix formatting</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/0b9c68f3b806ea097fe5f5cd01f49e6bd596e999">0b9c68f</a> Downgrade eslint to version used in NewDot</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/cc43cc006f76ef1bc4290844268e6101cf650c71">cc43cc0</a> Fix formatting</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/76ca522b6f3f39b9cb783e44cd4de3064042422d">76ca522</a> Add react-hooks/recommended</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/822601b25a838459293c291342ffe852b4b6bfc7">822601b</a> Merge pull request #84 from Expensify/OSBotify-bump-version-fa3ad8d2-c6b0-439c-afdb-89164e0f6505</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/4feab03c3e6c1c42ad6a5d37d75d37a2b103a00c">4feab03</a> 2.0.43</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/5155010f3410b9055dfcdf93a9dd2d4d6d7de78b">5155010</a> Merge pull request #83 from ishpaul777/lint/prohibit-default-props</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/f9a5a46369c3590df548a3c921d88364ff2cdfd6">f9a5a46</a> Merge branch &#x27;lint/prohibit-default-props&#x27; of github.com:ishpaul777/eslint-config-expensify into lint/prohibit-default-props</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/b516d74c30c36eee5d03997155c16aa53b041a36">b516d74</a> adds the lint rule</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/4b76fe8b0561baad7c498d2c81f214c07faafe4e">4b76fe8</a> adds the lint rule</li>
    </ul>

   <a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/compare/17dbfa4e063bfe50e94ebbc615e9d7dcf99203b0...72d77f0eae7e66775b98ded3e9c7f7b0a6144dc4">See the full diff</a>
  </details>
</details>

Check the changes in this PR to ensure they won't cause issues with your project.

------------

**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs.*

For more information:  
🧐 [View latest project report](https://app.snyk.io/org/expensify/project/35c87b83-0578-438a-9ce5-96d21d712614?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)

🛠 [Adjust project settings](https://app.snyk.io/org/expensify/project/35c87b83-0578-438a-9ce5-96d21d712614?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)

📚 [Read more about Snyk's upgrade and patch logic](https://support.snyk.io/hc/en-us/articles/360003891078-Snyk-patches-to-fix-vulnerabilities)

[//]: # (snyk:metadata:{"prId":"1483e750-ca17-4350-9b98-4597bd15e5f7","prPublicId":"1483e750-ca17-4350-9b98-4597bd15e5f7","dependencies":[{"name":"eslint-config-expensify","from":"2.0.42","to":"2.0.44"}],"packageManager":"npm","projectPublicId":"35c87b83-0578-438a-9ce5-96d21d712614","projectUrl":"https://app.snyk.io/org/expensify/project/35c87b83-0578-438a-9ce5-96d21d712614?utm_source=github&utm_medium=referral&page=fix-pr","type":"auto","patch":[],"vulns":["SNYK-JS-ANSIREGEX-1583908"],"upgrade":["SNYK-JS-ANSIREGEX-1583908"],"isBreakingChange":false,"env":"prod","prType":"fix","templateVariants":["updated-fix-title","priorityScore"],"priorityScoreList":[696],"remediationStrategy":"vuln"})

---

**Learn how to fix vulnerabilities with free interactive lessons:**

 🦉 [Regular Expression Denial of Service (ReDoS)](https://learn.snyk.io/lesson/redos/?loc&#x3D;fix-pr)

### CODE REVIEW

**Comment by mjasikowski** [comment]
Tested and working, but passing over to @tgolen for a quick review since there were minor syntax changes to satisfy new eslint-config rules.

File: src/js/component/list/List.js [hunk 1]
```diff
@@ -61,13 +61,6 @@ class List extends React.Component {
         });
     }
 
-    fetch() {
```
Comment (tgolen) [inline src/js/component/list/List.js:64]: How come this was removed?

**Comment by tgolen** [comment]
There are also conflicts on the lock file.

**Comment by mjasikowski** [comment]
I will resolve them after your review, I did merge other Snyk PRs in the meantime.

File: src/js/component/list/List.js (hunk 1 above)
Comment (mjasikowski) [reply src/js/component/list/List.js]: Strangely enough, after re-adding it's no longer complaining.

**Comment by tgolen** [comment]
Hm, I think something still isn't right with the lock file. The lint GH action is failing with this error:

```
npm error `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
```

File: src/js/component/list/List.js (hunk 1 above)
Comment (tgolen) [reply src/js/component/list/List.js:64]: Huh, interesting. Maybe it is safe to remove. I think as long as the K2 dashboard still loads all the data, it's probably all still working correctly.

Comment (tgolen) [reply src/js/component/list/List.js:64]: I think it would be better to remove the unused method instead of disabling the eslint rule for this line.

**Review by tgolen — CHANGES_REQUESTED** [review]
Glad you got the lock file working!

**Comment by mjasikowski** [comment]
Removed

**Review by tgolen — APPROVED** [review]

**Comment by tgolen** [comment]
I do not really think that it's necessary to bump the package.json version and deploy this PR. I think I'll just merge it and these changes can be silently rolled into the next version.

**Pull request merged**